File: SpeedTestGameClient/core/src/main/java/si/um/feri/speedii/classes/SessionManager.java
package si.um.feri.speedii.classes;

public class SessionManager {
    private boolean isSet = false;
    private String token = null;
    private User user = null;

    public boolean start(Pair<String, User> pair) {
        if (pair != null) {
            this.isSet = true;
            this.token = pair.getFirst();
            this.user = pair.getSecond();
            return true;
        }
        return false;
    }

    public void destroy() {
        this.isSet = false;
        this.token = null;
        this.user = null;
    }

    public boolean isSet() {
        return isSet;
    }

    public String getToken() {
        return token;
    }

    public User getUser() {
        return user;
    }

    @Override
    public String toString() {
        return "SessionManager{" +
            "isSet=" + isSet +
            ", token='" + token + '\'' +
            ", user=" + user +
            '}';
    }
}

File: MobileApp/app/speedtest/src/main/java/sosteric/pora/speedtest/SessionManager.kt

class SessionManager {
    var isSet: Boolean = false
    var token: String? = null
    var user: User? = null

    fun start(pair: Pair<String, User>?) : Boolean {
        if(pair != null){
            isSet = true
            token = pair.first
            user = pair.second
            return true
        }
        return false
    }

    fun destroy() {
        isSet = false
        token = null
        user = null
    }

    fun isLoggedIn() : Boolean {
        return isSet
    }

    override fun toString(): String {
        return "$isSet $token $user"
    }
}